zad2/warriors.rb

```diff
@@ -1,12 +1,75 @@
-class Warrior
+class Character
+  attr_accessor :name, :level
+
   def initialize(name:, level:)
+    @name = name
+    @level = validate_level(level) ? level : 1
+  end
+
+  def card
+    "#{@name} (lvl #{@level})"
   end
 
   def strength
+    [*1..12].sample + @level
+  end
+
+  private
+
+  def validate_level(level)
+    level.between?(1, 99)
+  rescue
```

Comment:
pamiętaj, że `rescue` bez podania konkretnego błędu wyłapie wszystkie błędy, co jest najczęściej niepożądanym działaniem. 

rozumiem, że `rescue` jest użyte przez możliwość podania czegoś co nie jest liczbą. Są jednak na to lepsze rozwiązania, np użycie metody `.to_i`:
```
validate_level(3) # => true
validate_level('3') # => true
validate_level('text') # => false
validate_level(nil) # => false
validate_level([]) # => NoMethodError
validate_level({}) # => NoMethodError
```
błędy pozostają przy podaniu tablicy, czy hasha, czyli już przy mocno ekstremalnych przypadkach.

Z drugiej strony. Czy jeśli użytkownik wpisze niewłaściwy typ czy nie powinniśmy odrazu go o tym informować zamiast przemilczać ten problem i udawać, że wszystko poszło ok? 🤔